pkg/utils/common.go

```diff
@@ -35,14 +35,32 @@ func Exists(filePath string) bool {
 
 // WaitUntil runs checkDone until a timeout is reached
 func WaitUntil(out io.Writer, poll, timeout time.Duration, checkDone func() error) error {
-	for start := time.Now(); time.Since(start) < timeout; time.Sleep(poll) {
+	_, err := timedRepeat(
+		out,
+		time.Now(),
+		poll,
+		timeout,
+		func(currentTime time.Time) time.Time {
+			time.Sleep(poll)
+			return time.Now()
+		},
+		checkDone)
+	return err
+}
+
+// timedRepeat runs checkDone until a timeout is reached by updating the current time via a specified operation
+func timedRepeat(out io.Writer, start time.Time, poll, timeout time.Duration, updater func(currentTime time.Time) time.Time, checkDone func() error) (time.Time, error) {
+	currentTime := start
+	endTime := currentTime.Add(timeout)
+
+	for ; currentTime.Before(endTime); currentTime = updater(currentTime) {
```

Comment:
Confused by this `for` syntax and what it is trying to tell me. I didn't know you could do this with the `;`, and it looks like an infinite loop to me. 

I guess it relies on the user `sleep`'ing, which is what we are going for here.

Reply:
It's basically like this go example: https://tour.golang.org/flowcontrol/2

No initializer, test = "current time < end time", update current time